README


This archive contains the distribution Tickit-DSL,
version 0.022:

  shortcuts for writing Tickit apps

This software is copyright (c) 2013 by Tom Molesworth.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Tickit-DSL

0.022     2014-08-31 15:07:05+01:00 Europe/London
	No new features.

	Dependencies:
	* More version updates

0.021     2014-08-31 03:03:34+01:00 Europe/London
	No new features.

	Dependencies:
	* More version updates

0.020     2014-08-31 01:25:41+01:00 Europe/London
	No new features.

	Dependencies:
	* Updated Tickit::Widget::Layout::Desktop and Menu/Tabbed

0.019     2014-08-31 00:37:46+01:00 Europe/London
	No new features.

	Dependencies:
	* Updated Tickit::Widget::Table.

0.018     2014-08-30 02:32:34+01:00 Europe/London
	No new features.

	Dependencies:
	* Updated some module versions

0.017     2014-08-23 02:45:55+01:00 Europe/London
	Features:
	* breadcrumb support (Tickit::Widget::Breadcrumb)
	* tree data parameter now handled by Tickit::Widget::Tree

	Bugs fixed:
	* Stray debugging line on table widget

0.016     2014-08-20 21:20:28+01:00 Europe/London
	Features:
	* tree accepts data parameter
	* float now gets the $float as first parameter
	* menu callbacks will be passed the container
	which holds the menubar, allowing floatbox-and-menu
	combinations (popup dialogs, for example)

0.015     2014-08-17 16:11:50+01:00 Europe/London
	Features:
	* desktop support (Tickit::Widget::Layout::Desktop)

0.014     2014-08-16 22:10:46+01:00 Europe/London
	Features:
	* console support (Tickit::Console)

	Documentation:
	* Fixed email address

0.013     2014-08-16 19:26:52+01:00 Europe/London
	Features:
	* table support (Tickit::Widget::Table)

0.012     2014-08-16 04:44:59+01:00 Europe/London
	Features:
	* fileviewer support (Tickit::Widget::FileViewer)

0.011     2014-08-06 16:50:56+01:00 Europe/London
	Features:
	* scroller_richtext support (Tickit::Widget::Scroller::Item::RichText)
	* floatbox and float support (Tickit::Widget::FloatBox)
	* Better support for button coderef - pass the container as the first
	parameter so that new widgets can be added consistently

0.010     2014-01-26 05:52:44+00:00 Europe/London
	Features:
	* Radio button support (Tickit::Widget::RadioButton)
	* Frame support (Tickit::Widget::Frame)
	* Dropped ::Table pending updated version on CPAN

0.009     2013-09-28 20:04:55 Europe/London
	Features:
	* Added checkbox (Tickit::Widget::CheckButton)
	* Added button (Tickit::Widget::Button)

	Dependencies:
	* Cleaned up cpanfile dep list

	Bug fixes:
	* Updated Dist::Zilla::Plugin::Test::Compile for RT88783

0.008     2013-09-15 19:01:51 Europe/London
	Features:
	* Support timers - 'timer { ... } after => 5;'

0.007     2013-09-15 16:40:15 Europe/London
	Features:
	* Support the ::Layout::Relative widget

	Bug fixes:
	* Make sure widget arguments are propagated where appropriate

	Documentation:
	* Add section headings to divide the widget functions by purpose

0.006     2013-09-14 23:57:33 Europe/London
	Features:
	* Support the ::Decoration widget

	Bug fixes:
	* Pass parameters to ::Placegrid (class, etc.)

0.005     2013-09-14 17:37:08 Europe/London
	Features:
	* Allow parent:X notation to avoid having to
	wrap everything in widget {} blocks.

0.004     2013-09-13 19:41:43 Europe/London
	Features:
	* Tickit::Widget::ScrollBox is available via scrollbox { }

0.003     2013-09-10 23:07:19 Europe/London
	No new features.

	Bug fixes:
	* gridrow was double-adding widgets. Now fixed.

0.002     2013-09-01 21:40:45 Europe/London
	Features:
	* Tickit::Widget::Statusbar is now on CPAN,
	so it has been added to the deps list

0.001     2013-09-01 03:41:06 Europe/London
	Initial CPAN release


examples/dsl.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Tickit::DSL;
use Eval::WithLexicals;
use Try::Tiny;
use Data::Dumper;

my $eval = Eval::WithLexicals->with_plugins("HintPersistence")->new(
	context => 'scalar',
);
vbox {
	my $scroller;
	widget {
		$scroller = scroller { };
	} expand => 1;
	entry {
		my ($self, $data) = @_;
		widget { scroller_text $data } parent => $scroller;
		try {
			my ($rslt) = $eval->eval($data);
			my $output = do {
				no warnings 'once';
				local $Data::Dumper::Terse = 1;
				local $Data::Dumper::Indent = 1;
				local $Data::Dumper::Useqq = 1;
				local $Data::Dumper::Deparse = 1;
				local $Data::Dumper::Sortkeys = 1;
				local $Data::Dumper::QuoteKeys = 0;
				Dumper($rslt)
			};
			widget { scroller_text $output } parent => $scroller
		} catch {
			my $err = $_;
			widget { scroller_text "Error: $err" } parent => $scroller
		}
	};
};
tickit->run;

=pod


  menubar {
    menuitem {
      menuitem { tickit->stop } label => 'Exit';
	} label => 'File';
    menuitem {
      menuitem { tickit->stop } label => 'Exit';
	} label => 'Widgets';
  } linetype => 'single';
  desktop { };
  statusbar { };

=cut



Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker 6.48;



my %WriteMakefileArgs = (
  "ABSTRACT" => "shortcuts for writing Tickit apps",
  "AUTHOR" => "Tom Molesworth <cpan\@perlsite.co.uk>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.48"
  },
  "DISTNAME" => "Tickit-DSL",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Tickit::DSL",
  "PREREQ_PM" => {
    "Exporter" => 0,
    "List::UtilsBy" => 0,
    "Tickit" => "0.46",
    "Tickit::Console" => "0.06",
    "Tickit::Widget::Breadcrumb" => "0.002",
    "Tickit::Widget::Decoration" => "0.004",
    "Tickit::Widget::FileViewer" => "0.004",
    "Tickit::Widget::FloatBox" => "0.02",
    "Tickit::Widget::Layout::Desktop" => "0.005",
    "Tickit::Widget::Layout::Relative" => "0.005",
    "Tickit::Widget::Menu" => "0.08",
    "Tickit::Widget::Progressbar" => "0.101",
    "Tickit::Widget::ScrollBox" => "0.03",
    "Tickit::Widget::Scroller" => "0.18",
    "Tickit::Widget::SegmentDisplay" => "0.02",
    "Tickit::Widget::SparkLine" => "0.104",
    "Tickit::Widget::Statusbar" => "0.004",
    "Tickit::Widget::Tabbed" => "0.016",
    "Tickit::Widget::Table" => "0.207",
    "Tickit::Widget::Tree" => "0.108",
    "Tickit::Widgets" => "0.19",
    "parent" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::CheckDeps" => "0.010",
    "Test::More" => "0.98"
  },
  "VERSION" => "0.022",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Exporter" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::UtilsBy" => 0,
  "Test::CheckDeps" => "0.010",
  "Test::More" => "0.98",
  "Tickit" => "0.46",
  "Tickit::Console" => "0.06",
  "Tickit::Widget::Breadcrumb" => "0.002",
  "Tickit::Widget::Decoration" => "0.004",
  "Tickit::Widget::FileViewer" => "0.004",
  "Tickit::Widget::FloatBox" => "0.02",
  "Tickit::Widget::Layout::Desktop" => "0.005",
  "Tickit::Widget::Layout::Relative" => "0.005",
  "Tickit::Widget::Menu" => "0.08",
  "Tickit::Widget::Progressbar" => "0.101",
  "Tickit::Widget::ScrollBox" => "0.03",
  "Tickit::Widget::Scroller" => "0.18",
  "Tickit::Widget::SegmentDisplay" => "0.02",
  "Tickit::Widget::SparkLine" => "0.104",
  "Tickit::Widget::Statusbar" => "0.004",
  "Tickit::Widget::Tabbed" => "0.016",
  "Tickit::Widget::Table" => "0.207",
  "Tickit::Widget::Tree" => "0.108",
  "Tickit::Widgets" => "0.19",
  "parent" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



